src/services/EventService.ts

```diff
@@ -50,4 +56,20 @@ export class EventService {
     const event = await this.eventRepository.findOne({ id });
     return event ? this.eventRepository.remove(event) : undefined;
   }
+
+  /**
+   * Creates an Attendance entity using the event obtained from eventId and
+   * the passed-in AppUser entity, then stores that Attendance entity to the
+   * Attendance table.
+   *
+   * @param eventId
+   * @param appUser
+   */
+  async registerAttendance(eventId: number, appUser: AppUser): Promise<Attendance> {
+    const event = await this.eventRepository.findOne({ id: eventId });
+    const attendance = this.attendanceService.createAttendance(event, appUser);
+    const savedAttendance = await this.attendanceService.saveAttendance(attendance);
```

Comment:
I don't have an answer to this question, but I think the current implementation allows for multiple sign-ins of the same user to the same event. they'll be stored in the database as the same thing (aka no problem of duplicated data) but there's no error code. perhaps in the future there is merit to "denying" a user multiple signins to the same event just from a UX perspective.

Reply:
that being said - this current implementation works and is good enough - just marking for historical purposes

Reply:
I think right now I allow updates to the Attendance entity. Since if the same user signs in again, it'd be the same composite PK so they'd just be updating their already existing Attendance entity. I think maybe we can move the updating to PUT (?) since at initial sign in, Attendance won't have values for fields like officer and duration.

Reply:
yeah i agree with the first part - probably PATCH makes more sense than PUT for updating in this specific case. what I was thinking was just that we might not want users to be able to make multiple POST calls to the attendance endpoint